src/App.tsx

```diff
@@ -1,35 +1,156 @@
+          >
+            <svg
+              className="w-6 h-6"
+              fill="none"
+              stroke="currentColor"
+              viewBox="0 0 24 24"
+              xmlns="http://www.w3.org/2000/svg"
+            >
+              {isMenuOpen ? (
+                <path
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                  strokeWidth="2"
+                  d="M6 18L18 6M6 6l12 12"
+                ></path>
+              ) : (
+                <path
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                  strokeWidth="2"
+                  d="M4 6h16M4 12h16M4 18h16"
+                ></path>
+              )}
+            </svg>
+          </button>
+        </div>
+
+        {/* Backdrop for mobile menu */}
+        {isMenuOpen && (
+          <div
+            className="fixed inset-0 bg-black bg-opacity-50 z-30 sm:hidden"
+            onClick={() => setIsMenuOpen(false)}
+          ></div>
+        )}
+
+        {/* Mobile menu / Desktop navigation */}
+        <ul
+          className={`
+            // Mobile styles
+            fixed inset-y-0 right-0 w-[70%] bg-white shadow-lg z-40
+            flex flex-col items-center justify-center space-y-6 pt-16
+            transform transition-transform duration-300 ease-in-out
+            ${isMenuOpen ? 'translate-x-0' : 'translate-x-full'}
+            ${!isMenuOpen && 'hidden'}
+        
+            // Override on sm and up (tablet and desktop)
+            sm:static sm:translate-x-0 sm:flex sm:flex-row sm:space-x-4 sm:space-y-0 sm:w-auto sm:bg-transparent sm:shadow-none sm:pt-0 sm:items-center sm:justify-end sm:z-auto
+          `}
+        >
+          <button
+            className="absolute top-4 right-4 text-gray-700 focus:outline-none sm:hidden"
+            onClick={() => setIsMenuOpen(false)}
+            aria-label="Close navigation"
+          >
+            <svg
+              className="w-6 h-6"
+              fill="none"
+              stroke="currentColor"
+              viewBox="0 0 24 24"
+              xmlns="http://www.w3.org/2000/svg"
+            >
+              <path
+                strokeLinecap="round"
+                strokeLinejoin="round"
+                strokeWidth="2"
+                d="M6 18L18 6M6 6l12 12"
+              ></path>
+            </svg>
+          </button>
+          {publicPages.map((page) => (
+            <li key={page}>
               <button
-                className="text-gray-700 font-semibold hover:text-green-700 transition"
-                onClick={() => setCurrentPage(page)}
+                className="text-gray-700 font-semibold hover:text-blue-600 transition text-lg sm:text-base px-4 py-2 rounded-md w-full text-center"
+                onClick={() => handlePageChange(page)}
               >
                 {page}
               </button>
             </li>
           ))}
+          {!isAuthenticated && (
```

Comment:
Your code would be cleaner if you move login and sign up together in the one if case:
```JS
{!isAuthenticated && (
  <>
    <li>...</li>
    <li>...</li>
  </>
)
}
```